search/api_test.py

```diff
@@ -0,0 +1,285 @@
+
+        assert_search(search(), [user.profile])
+
+    def test_education_update(self):
+        """
+        Test that Education is reindexed after being updated
+        """
+        user = UserFactory.create()
+        assert search()['total'] == 1
+        education = EducationFactory.create(profile=user.profile)
+        education.school_city = 'city'
+        education.save()
+
+        assert_search(search(), [user.profile])
+
+    def test_education_delete(self):
+        """
+        Test that Education is removed from index after being deleted
+        """
+        user = UserFactory.create()
+        education = EducationFactory.create(profile=user.profile)
+        assert_search(search(), [user.profile])
+        education.delete()
+        assert_search(search(), [user.profile])
+
+    def test_employment_add(self):
+        """
+        Test that Employment is indexed after being added
+        """
+        user = UserFactory.create()
+        assert search()['total'] == 1
+        EmploymentFactory.create(profile=user.profile)
+
+        assert_search(search(), [user.profile])
+
+    def test_employment_update(self):
+        """
+        Test that Employment is reindexed after being updated
+        """
+        user = UserFactory.create()
+        assert search()['total'] == 1
+        employment = EmploymentFactory.create(profile=user.profile)
+        employment.city = 'city'
+        employment.save()
+
+        assert_search(search(), [user.profile])
+
+    def test_employment_delete(self):
+        """
+        Test that Employment is removed from index after being deleted
+        """
+        user = UserFactory.create()
+        employment = EmploymentFactory.create(profile=user.profile)
+        assert_search(search(), [user.profile])
+        employment.delete()
+        assert_search(search(), [user.profile])
+
+    def test_remove_profile(self):
+        """
+        Test that remove_profile removes the profile from the index
+        """
+        user = UserFactory.create()
+        assert_search(search(), [user.profile])
+        remove_profile(user.profile)
+        assert_search(search(), [])
+
+    def test_index_profiles(self):
+        """
+        Test that index_profiles indexes an iterable of profiles
+        """
+        for _ in range(10):
+            with mute_signals(post_save):
+                profile = ProfileFactory.create()
+            # Not strictly necessary, the muted post_save will prevent indexing
+            remove_profile(profile)
+
+        # Confirm nothing in index
+        assert_search(search(), [])
+        index_profiles(Profile.objects.iterator(), chunk_size=4)
+        assert_search(search(), list(Profile.objects.all()))
+
+
+class SerializerTests(ESTestCase):
+    """
+    Tests for serializer
+    """
+
+    def test_profile_serializer(self):  # pylint: disable=no-self-use
+        """
+        Tests the profile serializer
+        """
```

Comment:
I still think it's a good idea for them to be mandatory, though this function is small enough to not need it. I'll reword this docstring to describe the test better